src/Plugin/Queue/Handlers/Source/DropSourceHandler.php

```diff
@@ -0,0 +1,152 @@
+ *                        alias: bool,
+ *                        base_expr: string,
+ *                        delim: bool
+ *                      }
+ *                   }
+ *               }
+ *           }
+ *       }
+ *   }>
+ */
+final class DropSourceHandler extends BaseDropHandler {
+
+	/**
+	 * @param string $name
+	 * @param string $tableName
+	 * @return int
+	 * @throws ManticoreSearchClientError
+	 */
+	protected function processDrop(string $name, string $tableName): int {
+		$manticoreClient = $this->manticoreClient;
+		$sql = /** @lang Manticore */
+			"SELECT * FROM $tableName WHERE match('@name \"$name\"')";
+
+
+		$result = $manticoreClient->sendRequest($sql);
+
+		if ($result->hasError()) {
+			throw ManticoreSearchClientError::create((string)$result->getError());
+		}
+
+		$removed = 0;
+		$result = $result->getResult();
+		if (is_array($result[0])) {
+			foreach ($result[0]['data'] as $sourceRow) {
+				$this->payload::$processor
+					->execute('stopWorkerById', [$sourceRow['full_name']]);
+
+				self::removeSourceRowData($sourceRow, $manticoreClient);
+				$removed++;
+			}
+		}
+
+
+		return $removed;
+	}
+
+	/**
+	 * @param array<string, string> $sourceRow
+	 * @param Client $client
+	 * @throws ManticoreSearchClientError
+	 */
+	protected static function removeSourceRowData(array $sourceRow, Client $client): void {
+
+		$queries = [
+			/** @lang Manticore */
+			"DROP TABLE {$sourceRow['buffer_table']}",
+			/** @lang Manticore */
+			"UPDATE _views SET suspended=1 WHERE match('@source_name \"{$sourceRow['full_name']}\"')",
+			/** @lang Manticore */
+			"DELETE FROM _sources WHERE id = {$sourceRow['id']}",
+		];
+
+		foreach ($queries as $query) {
+			$request = $client->sendRequest($query);
+			if ($request->hasError()) {
+				throw ManticoreSearchClientError::create((string)$request->getError());
+			}
+		}
+	}
+
+	/**
+	 * @param Payload<array{
+	 *       DROP: array{
+	 *           expr_type?: string,
+	 *           option: bool,
+	 *           if-exists: bool,
+	 *           sub_tree: array{
+	 *               array{
+	 *                   expr_type: string,
+	 *                   base_expr: string
+	 *               },
+	 *               array{
+	 *                   expr_type: string,
+	 *                   base_expr: string,
+	 *                   sub_tree: array{
+	 *                      array{
+	 *                        expr_type: string,
+	 *                        table: string,
+	 *                        no_quotes: array{
+	 *                            delim: bool,
+	 *                            parts: array<string>
+	 *                        },
+	 *                        alias: bool,
+	 *                        base_expr: string,
+	 *                        delim: bool
+	 *                      }
+	 *                   }
+	 *               }
+	 *           }
+	 *       }
+	 *   }> $payload
+	 * @return string
+	 */
+	#[\Override] protected function getName(Payload $payload): string {
```

Comment:
What Override is used here fore? We never used it before and I do not think that we really need it, cuz there is nothing in code that parse meta attributes by using reflection